src/Common.am

```diff
@@ -73,3 +73,7 @@ COMPAT_LIB = $(top_builddir)/compat/libcompatsquid.la $(LIBPROFILER)
 
 ## Some helpers are written in Perl and need the local shell defined properly
 subst_perlshell = sed -e 's,[@]PERL[@],$(PERL),g' <$(srcdir)/$@.pl.in >$@ || ($(RM) -f $@ ; exit 1)
+
+EXTENSIONS=.hdrtest
+.h.hdrtest:
+	$(SHELL) $(top_srcdir)/test-suite/testheaders.sh "$(CXXCOMPILE)" $^ || exit 1
```

Comment:
My primary concern is that the corresponding code in testheaders.sh cannot work correctly in "parallel" header tests because it will (AFAICT) mark the whole testHeaders target completed even if some headers fail the test.

```sh
test $exitCode -eq 0 && cp ${TRUE} testHeaders
```

To avoid extra work, my other change requests may be ignored until this primary concern is addressed.

Reply:
https://bugs.squid-cache.org/show_bug.cgi?id=5237